cameraview/src/main/java/com/otaliastudios/cameraview/engine/Camera2Engine.java

```diff
@@ -1252,6 +1261,33 @@ public void setPlaySounds(boolean playSounds) {
         mPlaySoundsOp.end(null);
     }
 
+    @Override public void setPreviewFrameRate(float previewFrameRate) {
+        mPreviewFrameRate = previewFrameRate;
+        mHandler.run(new Runnable() {
+            @Override
+            public void run() {
+                if (getEngineState() == STATE_STARTED) {
+                    if (applyPreviewFrameRate(mRepeatingRequestBuilder)) {
+                        applyRepeatingRequestBuilder();
+                    }
+                }
+            }
+        });
+    }
+
+    @SuppressWarnings("WeakerAccess")
+    protected boolean applyPreviewFrameRate(@NonNull CaptureRequest.Builder builder) {
+        Collection<Range<Integer>> fpsRange = mCameraOptions.getSupportedFpsRange();
+        for (Range<Integer> range : fpsRange) {
+            if (range.getUpper() >= mPreviewFrameRate) {
+                mIsHigherFrameRateSupported = true;
+                builder.set(CaptureRequest.CONTROL_AE_TARGET_FPS_RANGE, range);
+                return true;
+            }
+        }
+        return false;
```

Comment:
Also before returning `false`, you should reset `mPreviewFrameRate` to the old value like applyHdr for example